src/PSR7/SpecFinder.php

```diff
@@ -74,37 +108,49 @@ public function findPathSpec(OperationAddress $addr) : PathItem
         return $pathSpec;
     }
 
-    /**
-     * @return Parameter[]
-     *
-     * @throws NoPath
-     */
-    public function findPathSpecs(OperationAddress $addr) : array
+    private function findPathSimple(OperationAddress $addr) : ?PathItem
     {
-        $spec = $this->findOperationSpec($addr);
+        return $this->openApi->paths->getPath($addr->path());
+    }
 
-        // 1. Collect operation-level params
-        $pathSpecs = [];
+    private function findPathUsingFinder(OperationAddress $addr) : ?PathItem
+    {
+        $finder  = new PathFinder($this->openApi, $addr->path(), $addr->method());
+        $results = $finder->search();
+        if (count($results) === 1) {
+            return $this->findPathSimple($results[0]);
+        }
 
-        foreach ($spec->parameters as $p) {
-            if ($p->in !== 'path') {
-                continue;
-            }
+        return null;
```

Comment:
Another problem from original PR. Returning null from here is processed with `NoPath` exception while it could be more than one matching

Reply:
Agreed, what's your suggestion.
1. Ambiguous exception - in case of count>1
2. Return first match.
3. Apply a ruleset to determine, which spec might be the right one?

Maybe we start with (2) and think about a more mature solution in upcoming releases.

Reply:
Solution 2 for now - return first match